flink-python/src/main/java/org/apache/flink/python/env/beam/ProcessPythonEnvironmentManager.java

```diff
@@ -326,9 +327,21 @@ private void constructArchivesDirectory(Map<String, String> env) throws IOExcept
 
             // extract archives to archives directory
             for (Map.Entry<String, String> entry : dependencyInfo.getArchives().entrySet()) {
-                ZipUtils.extractZipFileWithPermissions(
-                        entry.getKey(),
-                        String.join(File.separator, archivesDirectory, entry.getValue()));
+                String filePath = entry.getKey();
+                if (filePath.endsWith(".zip") || filePath.endsWith(".jar")) {
```

Comment:
It's better to convert filePath to lowercase and then checks the suffix

Reply:
It would be great if we could support windows as I found that there are still quite a few users using window for development.

Reply:
done